README.md

```diff
@@ -4,8 +4,153 @@ Measurement library will provide an open source alternative for sites/apps to se
+sent events using a variety of built in event processors (e.g. sending data to Google Analytics).
+
+Data about the user is automatically saved to persist between visits using your choice of
+a variety of different storage options (e.g. cookies).
+
+# Installation
+First, install the package with your favorite package manager:
+```bash
+yarn add measurement-library
+// or
+npm install measurement-library
+```
+
+Next, choose [the event processor](#event-processors) and [the storage implementation](#storage-interfaces)
+that you want to use and include this code in your `<head>` tag.
+```html
+<script async src="./node_modules/measurement-library/dist/measure"/>
+<script>
+  window.dataLayer = window.dataLayer || [];
+  function measure(){dataLayer.push(arguments);}
+  // One or more config commands. To send data to google analytics while 
+  // using cookies for storage:
+  measure('config', 'googleAnalytics', {}, 'cookies', {});
+</script>
+```
+
+[//]: # (TODO: kj Make the config command use the right settings for google analytics)
+
+For the development version (bigger file size, but reports possible errors to the console):
+```html
+<script async src="./node_modules/measurement-library/dist/measure-debug"/>
+<script>/* Configure script like last example. */ </script>
+```
+
+# Usage
+After setting up the script tags, you can begin to use the library. All commands
+are processed by passing arguments to the  `measure()` function. 
+
+## Event command
+When you run the command `measure('event', eventName, eventOptions)`, all registered
+[event processors](#event-processors) will read the event and perform actions corresponding
+to their API.
+
+## Set command
+To save parameters beyond the current page, you can call the command `measure('set', key, value)`.
+The registered event processor will determine if the value should be saved, how long the value should
+be saved for, and save it. To overwrite this behavior, you can specify a third time-to-live parameter:
+`measure('set', key, value, secondsToLive)`. In particular, if `secondsToLive` is 0, no data will be saved
+to long term storage.
+
+The set command can also be used to set parameters related to a implementation of
+an event processor or storage interface. For example, to modify the default parameters
+of the cookies storage, run a `set` command in the script tag before calling
+`config`. 
+
+```js
+// Set the default cookie parameters
+measure('set', 'cookies', {prefix: 'my_', expires: 11});
+// Use the default cookie parameters: prefix 'my_' and expires 11.
+measure('config', 'eventProcessorName', {}, 'cookies', {});
+// Override a default cookie parameter: expires is 22 for this storage.
+measure('config', 'eventProcessorName', {}, 'cookies', {expires: 22});
+```
+
+# Event Processors
+## Google Analytics
+ [//]: # (TODO: kj)
+
+## Custom Event Processor
+If none of the above fit your needs, you can create your own event processor.
```

Comment:
Took me a second to grok the scope of "none of the above" (since there's a lot of doc above this line :)). Something a bit more specific, like "If none of the above *built-in processors* fit...."